conans/client/tools/win.py

```diff
@@ -14,10 +14,87 @@
 from conans.util.env_reader import get_env
 from conans.util.files import decode_text, save, mkdir_tmp
 from conans.unicode import get_cwd
+from conans.model.version import Version
 
 _global_output = None
 
 
+def _visual_compiler_cygwin(output, version):
+    if os.path.isfile("/proc/registry/HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Windows/CurrentVersion/ProgramFilesDir (x86)"):
+        is_64bits = True
+    else:
+        is_64bits = False
+
+    if is_64bits:
+        key_name = r'HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\VisualStudio\SxS\VC7'
+    else:
+        key_name = r'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\VisualStudio\SxS\VC7'
+
+    if not os.path.isfile("/proc/registry/" + key_name.replace('\\', '/') + "/" + version):
+        return None
+
+    installed_version = Version(version).major(fill=False)
+    compiler = "Visual Studio"
+    output.success("CYGWIN: Found %s %s" % (compiler, installed_version))
+    return compiler, installed_version
+
+
+@contextmanager
```

Comment:
Remove decorator and I will add a small docstring describing the parameters of the function, specially for `key` being a `winreg` key from the `six.moves` module

Reply:
done